### ISSUE TITLE
Blink and Brightness mechanisms not exposed

### ISSUE THREAD

**Issue by TimOddy** [pimoroni/fourletter-phat#4]
The blink and brightness and mechanisms; "fourletterphat.set_blink(frequency)" and "fourletterphat.set_brightness(brightness)" are not usefully exposed for users.  Currently the module returns an error if you try and use them.  They seem to be in HT16K33.py, but not correctly referenced to be seen using "fourletterphat" alone, as given in the examples (none of which actually use blinking or brightness).

__init__.py needs this added

```
set_brightness = display.set_brightness
set_blink = display.set_blink
```

alphanum4.py needs something like this add

```
   def set_brightness(self,bt):
        self.set_bright(bt)

    def set_blink(self,bl):
        self.set_blnk(bl)
```

(and consequently set_brightness and set_blink in HT16K33.py need to be renamed.  Using the same name appears to cause an infinite loop.  I'm not experienced with Python Modules, or evenPython (!), so there may be a better way to do this).

**Comment by Gadgetoid** [comment]
Good spot. You're right, these should probably be added at least for completeness. I think I opted to skip `set_blink` because I couldn't envision anyone using it, but that's not necessarily a good reason to omit a feature that's already there.

Because `AlphaNum4` inherits directly from `HT16K33` there's no need to define those methods again within `alphanum4.py`, they already implicitly exist through inheritance:

```python
class AlphaNum4(HT16K33.HT16K33)
```

That means you should just be able to change `init.py` with the two lines you mention, and nothing else.

The inheritance is why you're seeing an infinite loop- because you're overriding the `set_brightness` and `set_blink` methods, they're simply calling themselves, rather than the ones in the inherited class.

If you *wanted* to override them (sometimes there's good reason to do so) then you must explicitly call the versions on the parent class, like so:

```python
def set_brightness(self,bt):
    HT16K33.HT16K33.set_brightness(self, bt)

def set_blink(self,bl):
    HT16K33.HT16K33.set_blink(self, bl)
```

Which is basically calling the methods from the definition of `HT16K33` and passing it an instance of the child class to operate upon.

In new-style Python classes, which inherit from `object`, you can also accomplish this without explicitly calling the parent class by name:

```python
def set_brightness(self,bt):
    super(AlphaNum4, self).set_brightness(bt)

def set_blink(self,bl):
    super(AlphaNum4, self).set_blink(bl)
```

This very pattern, in fact, is used in the `__init__` method of `AlphaNum4`, so copying this pattern would be the correct thing to do in this case (although in my opinion the first method I mentioned is far more concise and easy to understand what's going on):

```python
class AlphaNum4(HT16K33.HT16K33):
    """Alphanumeric 14 segment LED backpack display."""

    def __init__(self, **kwargs):
        """Initialize display.
        All arguments will be passed to the HT16K33 class initializer,
        including optional I2C address and bus number parameters.
        """

        super(AlphaNum4, self).__init__(**kwargs)
```

**Comment by Gadgetoid** [comment]
These methods have been added as of: https://github.com/pimoroni/fourletter-phat/commit/b31c16095b7f6f2fcd491c92988975ed65c9c06c

And should be available in release 0.0.2 when we publish it!
